src/axom/primal/operators/clip.hpp

```diff
@@ -98,6 +99,86 @@ Polygon<T, 3> clip(const Triangle<T, 3>& tri, const BoundingBox<T, 3>& bbox)
   return *currentPoly;
 }
 
+/*!
+ * \brief Clips a 3D hexahedron against a tetrahedron in 3D, returning
+ *        the geometric intersection of the hexahedron and the tetrahedron
+ *        as a polyhedron
+ *
+ *  This function clips the hexahedron by the 4 planes obtained from the
+ *  tetrahedron's faces (normals point inward). Clipping the
+ *  hexahedron/polyhedron by each plane gives the polyhedron above that plane.
+ *  Clipping the polyhedron by a plane involves
+ *  finding new vertices at the intersection of the polyhedron edges and
+ *  the plane, removing vertices from the polyhedron that are below the
+ *  plane, and redefining the neighbors for each vertex (a vertex is a
+ *  neighbor of another vertex if there is an edge between them).
+ *
+ *
+ * \param [in] hex The hexahedron to clip
+ * \param [in] tet The tetrahedron to clip against
+ * \param [in] eps The epsilon value
+ * \param [in] checkSign If true (default is false), checks the volume of the
+ *             shapes are positive. If volume is negative, order of some
+ *             vertices will be swapped.
+ *
+ * \return A polyhedron of the hexahedron clipped against the tetrahedron.
+ *
+ * \note Function is based off clipPolyhedron() in Mike Owen's PolyClipper.
```

Comment:
Minor: Should we expose `clipPolyhedron` to users? Or perhaps `clip(Polyhedron, Polyhedron)`? w/ perhaps some constraints on the shape of the second polyhedron, e.g. that it is composed of planar facets and is convex?

If so, I'd suggest adding it in a follow-up PR. We might want to wait until there is a need/request for more general shapes.

Reply:
Captured in an issue here: #1068.